inspirehep/modules/records/api.py

```diff
@@ -90,6 +91,56 @@ def _create_bucket(self, location=None, storage_class=None):
         )
         return bucket
 
+    def attach_file(self, file_name, file):
```

Comment:
I think the name of the function is misleading, as it's not really attaching a file, it's actually fixing the current record to be consistent. I would definetly try to add a file to the record without adding it to the documents or figures first...

Hmmm...

Maybe it's nicer to have a function in the record like `reconcile_documents_and_figures_keys` that iterates over the current documents and figures and downloads whatever it has adding the recid to the keys and such, then that one can use an internal function (this one) renamed to something like `_download_to_file` or something similar that does the download to the files property.